lsql/judge/views.py

```diff
@@ -323,3 +326,33 @@ def test_error_500(request):
         elems = list()
         return HttpResponse(elems[55])  # list index out of range, error 500
     return HttpResponseNotFound()
+
+
+def filter_expected_db(expected_db, initial_db):
+    """Compare expected_db and initial_db and return all the modified, removed or added tables"""
+    expected_tables = sorted(list(expected_db.keys()))
+    initial_tables = sorted(list(initial_db.keys()))
+    add_list = []
+    modified_list = []
+    remove_list = []
+    ret_added = {}
+    ret_modified = {}
+    ret_removed = {}
+    if expected_tables != initial_tables:
+        add_list = [x for x in expected_tables if x not in initial_tables]
+        remove_list = [y for y in initial_tables if y not in expected_tables]
+        for element in add_list:
+            expected_tables.remove(element)
+        for element in remove_list:
+            initial_tables.remove(element)
+    for table in expected_tables:
+        veredict, _ = compare_select_results(expected_db[table], initial_db[table], order=False)
+        if veredict != VeredictCode.AC:
+            modified_list.append(table)
+    for item in add_list:
```

Comment:
Ya que sabes usar listas intensionales, puedes crear directamente ret_modified, ret_added y red_removed como un **diccionario definido de manera intensional** y queda mucho más compacto. Ejemplo:

    `{str(i): i for i in range(1000) if i %2 == 0}`

Reply:
Entiendo que te refieres a algo así, pero me da error de KeyError, no sé muy bien por qué.

`{ret_modified[item]:expected_db[item] for item in modified_list}`

Reply:
![image](https://user-images.githubusercontent.com/48255774/110352547-581fcb00-8036-11eb-9d50-6a6c88878b8c.png)

Reply:
Sí, algo parecido aunque creo que te puedes ahorrar la lista `modified_list`. `KeyError` puede aparecer porque la clave `item` no existe en `ret_modified`. Y ten cuidado porque las claves de un diccionario no pueden ser mutables. ¿Qué tipo tiene `ret_modified[item]`? Si es otro diccionario no vale.

Reply:
Es que no existe item en ret_modified porque lo estoy asignando con la iteración. Para poder hacer una lista intensional como me dices debo declarar primero el objeto item y poder asignarlo correctamente al diccionario. En ese punto ret_modified está complemente vacío.

Reply:
El diccionario `{ret_modified[item]:expected_db[item] for item in modified_list}` lo creas para asignarlo a `ret_modified`, ¿no? Claramente no puedes usarlo en la definición. Lo de la izquierda de los puntos es la clave. ¿Qué quieres que sea la clave? ¿Un str con el nombre de la tabla? ¿Un diccionario como parece que tienes ahora?

Reply:
Perdona, no había comprendido la función, hoy estoy un poco espero. Ya he entendido y ya lo he hecho.